src/lib/Simplify.hs

```diff
@@ -151,6 +149,13 @@ simplifyExpr expr = confuseGHC >>= \_ -> case expr of
             liftM Var $ emit $ Case e' alts' resultTy' eff'
           False -> defuncCase e' alts resultTy'
 
+caseComputingEffs
+  :: forall m n. (MonadFail1 m, EnvReader m)
+  => Atom n -> [Alt n] -> Type n -> m n (Expr n)
+caseComputingEffs scrut alts resultTy = do
+  effs <- forM alts effectsE
```

Comment:
you might be able to `foldMapM` here. I think we define that in Utils. Then this whole function becomes
```
Case scrut alts resultTy <$> foldMapM effectE alts
```

Reply:
Done.